kolibri/content/management/commands/content.py

```diff
@@ -0,0 +1,156 @@
+        )
+        movedir_subparser.add_argument(
+            'destination',
+            type=str,
+            help='Destination of the new content directory.'
+        )
+
+    def handle(self, *args, **options):
+        if options['command'] == 'movedirectory':
+            self.move_content_directory(options['destination'])
+
+    def move_content_directory(self, dst):
+        # Check exsitence of current directory before running the commands.
+        content_directory = OPTIONS['Paths']['CONTENT_DIR']
+        destination = os.path.abspath(os.path.expanduser(dst))
+
+        if not os.path.exists(content_directory):
+            self.stderr.write(
+                self.style.ERROR(
+                    '\nCurrent content directory {} does not exist.'
+                ).format(content_directory))
+            raise SystemExit()
+
+        # If the current directory is the same as the destination
+        if content_directory == destination:
+            self.stderr.write(
+                self.style.ERROR(
+                    '\nDestination {} is the current content directory.'
+                ).format(destination))
+            raise SystemExit()
+
+        # Check if Kolibri is running
+        try:
+            server.get_status()
+            self.stderr.write(self.style.ERROR(
+                "Cannot migrate content while Kolibri is running, please run:\n"
+                "\n"
+                "    kolibri stop\n"
+            ))
+            raise SystemExit()
+        except server.NotRunning:
+            pass
+
+        self.migrate(content_directory, destination)
+
+    def migrate(self, src, dst):
+        """
+        Migrate the content from current content directory to the destination.
+        """
+        logger.info('Current content directory is {}'.format(src))
+        logger.info('Migrating the content into {}'.format(dst))
+
+        databases_src = os.path.join(src, 'databases')
+        databases_dst = os.path.join(dst, 'databases')
+        storage_src = os.path.join(src, 'storage')
+        storage_dst = os.path.join(dst, 'storage')
+
+        # Check if destination has content by checking if databases folder is not empty
+        dst_content_exists = os.path.exists(databases_dst) and \
+            os.listdir(databases_dst)
+
+        # If destination has content inside, ask users if they want to keep the content
+        if dst_content_exists:
+            self.handle_user_input(src, dst)
+
+        # copy the databases folder
+        self.copy_content(databases_src, databases_dst)
+        shutil.rmtree(databases_src)
+
+        # copy the storage folder only when the source folder contains it
+        if os.path.exists(storage_src):
+            self.copy_content(storage_src, storage_dst)
+            shutil.rmtree(storage_src)
+
+        self.update_config_content_directory(dst)
+
+    def handle_user_input(self, src, dst):
```

Comment:
This method is a bit too generic in its naming. But you will also be able to significantly reduce the complexity by inverting the question. So instead of asking "do you want to keep x?", try asking:

> Destination already exists: /x/y/z
>
> Do you want to overwrite it completely? (y/N)

...then refactor to match.

Notice that the capital letter is the default answer that occurs by pressing ENTER, this is a convention of some kind.

Reply:
thank you for the information! I will change the name and the question